fava/serialisation.py

```diff
@@ -43,14 +47,66 @@ def extract_tags_links(string):
     return new_string, frozenset(tags), frozenset(links)
 
 
-def parse_number(num):
-    """Parse a number as entered in an entry form, supporting division."""
-    if not num:
+def _is_float_literal(literal):
+    return re.search(r"^[0-9.][0-9.]*$", literal)
+
+
+def _decimalize_statement(statement):
+    """Wraps every float number in the statement string with a call to D().
+
+    Example: "3.14" -> "D(3.14)"
+    """
+    result = []
+    tokens = generate_tokens(StringIO(statement).readline)
+    for toknum, tokval, _, _, _ in tokens:
+        if toknum == NUMBER and _is_float_literal(tokval):
```

Comment:
Is the second part really needed? Can't we just assume that we can wrap any number (that is sensible input here) in a call to D()?

Reply:
Wouldn't it be simpler overall to replace this function with one that just does a simple regular expression replace from number to D('number')?

Reply:
Changed to any number. However, I'd prefer to leave the discovery of numbers to Python, rather than trying to craft a custom regexp.